ice-rest-catalog/src/test/java/com/altinity/ice/rest/catalog/RESTCatalogIT.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2025 Altinity Inc and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package com.altinity.ice.rest.catalog;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import com.altinity.ice.internal.strings.Strings;
+import com.altinity.ice.rest.catalog.internal.config.Config;
+import com.altinity.ice.rest.catalog.internal.etcd.EtcdCatalog;
+import com.google.common.net.HostAndPort;
+import io.etcd.jetcd.KV;
+import io.etcd.jetcd.Txn;
+import org.apache.iceberg.inmemory.InMemoryFileIO;
+import org.eclipse.jetty.server.Server;
+import org.testcontainers.containers.GenericContainer;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import picocli.CommandLine;
+
+import static com.altinity.ice.rest.catalog.Main.createAdminServer;
+import static com.altinity.ice.rest.catalog.Main.createServer;
+
+public class RESTCatalogIT {
+
+  private Thread serverThread;
+
+  private EtcdCatalog catalog;
+  private Consumer<KV> preKvtx;
+  private Server httpServer;
+  private Server adminServer;
+
+  @SuppressWarnings("rawtypes")
+  private final GenericContainer etcd =
```

Comment:
lets move RESTCatalogIT to a separate PR. There are still a few things to iron out here:
- doesn't looks like we really need etcd here. in-memory sqlite should be enough
- redundant icebergConfig in https://github.com/Altinity/ice/pull/26/files#diff-ca269f882cb860088e50c407fa428d338e81103552eedf6376bcca8768fbab18R76-R84
- Thread.sleep(5000);
- not sure how https://github.com/Altinity/ice/pull/26/files#diff-ca269f882cb860088e50c407fa428d338e81103552eedf6376bcca8768fbab18R124-R126 can possibly work considering that we're not starting minio here
- adminServer is not needed